client/state/themes/actions.js

```diff
@@ -44,7 +44,7 @@ export function fetchThemes( site ) {
 
 		debug( 'Query params', queryParams );
 
-		wpcom.undocumented().themes( site, queryParams )
+		return wpcom.undocumented().themes( site, queryParams )
```

Comment:
Returning the promise from the data fetch lets us respond after asynchronous dispatches complete. This is necessary for server-rendering so the server doesn't respond before our asynchronous fetch completes.